### PR TITLE
introduce memory limits (in tests)

### PR DESCRIPTION
**Pull request by jtnord**
whilst running tests I experienced some random memory issues.

As there was no attempt at running maven / surefire or even the JVM spawned inside a test with amy memory limits I added some.

The memory chosen may or may not be optimal, no attempt has been made to optimize the values.  they have mostly be taken from other places that are known to be reasonable.

### Testing done

`mvn deploy`

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: .mvn/jvm.config [hunk 1]
```diff
@@ -0,0 +1 @@
+-Xmx256m -XX:+HeapDumpOnOutOfMemoryError -XX:+TieredCompilation -XX:TieredStopAtLevel=1
```
Comment (jtnord) [inline .mvn/jvm.config:1]: maven itself

File: pom.xml [hunk 2]
```diff
@@ -65,6 +65,7 @@ THE SOFTWARE.
     <spotbugs.effort>Max</spotbugs.effort>
     <spotbugs.threshold>Low</spotbugs.threshold>
     <bc-version>1.78.1</bc-version>
+    <argLine>-Xms256M -Xmx256M -XX:+TieredCompilation -XX:TieredStopAtLevel=1</argLine>
```
Comment (jtnord) [inline pom.xml:68]: surefire.  omitting `-XX:+HeapDumpOnOutOfMemoryError` as we have at least one test that intentionally introduces an memory exhastion.

File: src/test/java/hudson/remoting/ForkRunner.java [hunk 3]
```diff
@@ -22,6 +22,7 @@ protected List<String> buildCommandLine() {
         String cp = getClasspath();
 
         List<String> r = new ArrayList<>();
+        r.add("-Xmx128M");
```
Comment (jtnord) [inline src/test/java/hudson/remoting/ForkRunner.java:25]: remoting JVM for tests

**Review by timja — APPROVED** [review]
if it works

**Comment by jtnord** [comment]
> if it works

CI Passed :-)

**Pull request merged**